rai/node/node.cpp

```diff
@@ -1303,7 +1303,14 @@ void rai::block_processor::process_receive_many (std::unique_lock<std::mutex> &
 			forced.pop_front ();
 			force = true;
 		}
+		/* Verify more state blocks if blocks deque is empty
+		Because verification is long process, avoid large deque verification inside of write transaction */
+		bool start_verification (blocks.empty () && !state_blocks.empty () && state_blocks.size () < 2048);
 		lock_a.unlock ();
+		if (start_verification)
+		{
+			verify_state_blocks (lock_a);
```

Comment:
Instead of making it conditional upon being below a certain threshold, can we instead modify `verify_state_blocks()` to accept a maximum number of blocks to verify (defaulting to all).

That way we maximally be within our quanta for the DB transaction (+ up to verification spill).

```suggestion
			verify_state_blocks (lock_a, 2048);
```

With the appropriate updates to `verify_state_blocks()`

Reply:
Proposed `verify_state_blocks()` looks something like:

```
void rai::block_processor::verify_state_blocks (std::unique_lock<std::mutex> & lock_a, _sizetype_ max_count = ...::limits<inf>)
{
	lock_a.lock ();
	std::deque<std::pair<std::shared_ptr<rai::block>, std::chrono::steady_clock::time_point>> items;
	if (max_count == ...::limits<inf> || max_count >= state_blocks.size ())
	{
		items.swap (state_blocks);
	}
	else
	{
		auto keep_size (state_blocks.size () - max_count);
		items.resize (keep_size);
		std::swap_ranges (state_blocks.end () - keep_size, state_blocks.end (), items.begin ());
		state_blocks.resize (max_count);
		items.swap (state_blocks);
	}
	lock_a.unlock ();
...
```